storage/spanstore/metrics/decorator.go

```diff
@@ -72,11 +75,42 @@ func buildQueryMetrics(operation string, metricsFactory metrics.Factory) *queryM
 	return qMetrics
 }
 
+func buildQueryMetricsWithTags(operation string, tags map[string]string, metricsFactory metrics.Factory) *queryMetrics {
+	qMetrics := &queryMetrics{}
+	mergedTags := map[string]string{"operation": operation}
+	for k, v := range tags {
+		mergedTags[k] = v
+	}
+	scoped := metricsFactory.Namespace(metrics.NSOptions{
+		Name: "",
+		Tags: mergedTags,
+	})
+	metrics.Init(qMetrics, scoped, nil)
+	return qMetrics
+}
+
 // FindTraces implements spanstore.Reader#FindTraces
 func (m *ReadMetricsDecorator) FindTraces(ctx context.Context, traceQuery *spanstore.TraceQueryParameters) ([]*model.Trace, error) {
 	start := time.Now()
 	retMe, err := m.spanReader.FindTraces(ctx, traceQuery)
-	m.findTracesMetrics.emit(err, time.Since(start), len(retMe))
+	search_window_buckets := []string{"0s", "5m", "15m", "1h", "3h", "6h", "12h", "1d", "3d"}
+
+	search_window_duration := traceQuery.StartTimeMax.Sub(traceQuery.StartTimeMin).Microseconds()
+	search_window_bucket := ""
+
+	// Go through the buckets in reverse
+	for i := len(search_window_buckets) - 1; i >= 0; i-- {
+		if search_window_duration >= func() time.Duration { t, _ := time.ParseDuration(search_window_buckets[i]); return t }().Microseconds() {
+			search_window_bucket = search_window_buckets[i]
+		}
+	}
+
+	tags := map[string]string{
+		"service":       traceQuery.ServiceName,
+		"results_limit": strconv.Itoa(traceQuery.NumTraces),
```

Comment:
nit: it seems the UI uses `Limit Results` for this parameter, maybe we could align the name.

```suggestion
		"limit_results": strconv.Itoa(traceQuery.NumTraces),
```